components/datetime/src/format/datetime.rs

```diff
@@ -155,12 +161,17 @@ where
                 loc_datetime,
                 fixed_decimal_format,
                 w,
-            )?,
+            )?
+            .unwrap_or_else(|e| {
+                if r.is_ok() {
+                    r = Err(e);
+                }
```

Comment:
Very optional, here and elsewhere:

```suggestion
                r = r.and(Err(e))
```

Reply:
ah nice